### DISCUSSION TITLE
POST_INIT commands cause boot failure in LTSP client boot

### DISCUSSION THREAD

**Discussion by mavrinac** [ltsp/ltsp#800] [General]
This just started happening this morning (I just got back from a vacation!)

Any POST_INIT commands/sections in /etc/ltsp/ltsp.conf fail to evaluate on client boot.
If they are commented out, the clients boot.

This is a problem because some clients need customization.
1. I don't want any users to see the nm-applet on the clients.
2. The server has nvidia drivers loaded, but many of the clients need to load the amdgpu drivers (which are already installed on the server) for the X-server to start.

These are the relevant sections included in ltsp.conf:

[client]
...
POST_INIT_LAB_CLIENT="section_lab_client"

[04:7b:cb:*]
POST_INIT_RM_BLACKLIST_AMDGPU="section_rm_blacklist_amdgpu"

[lab_client]
rm /etc/xdg/autostart/nm-applet.desktop

[rm_blacklist_amdgpu]
rm /etc/modprobe.d/blacklist-amdgpu.conf
modprobe amdgpu

**Comment by alkisg** [comment]
Hi, paste your whole ltsp.conf (you may change whatever sensitive information it has),
and a photo of the client screen when it fails to boot.

**Reply by mavrinac** [reply]
/etc/ltsp/ltsp.conf

[server]
IPXE_X86_64_IMG="Debian Bookworm"

[common]
MENU_TIMEOUT=30000

[clients]
CRONTAB_01="0 6 * * * root /sbin/shutdown -h now"
CRONTAB_02="0 9,3 * * * root ntpdate xxx.xxx.xxx.xxx; /usr/sbin/hwclock --systohc"
CUPS_SERVER=xxx.xxx.xxx.xxx
#DEBUG_LOG=1
#DEBUG_SHELL=1
KERNEL_PARAMETERS="security=none"
#DISABLE_SESSION_SERVICES=""
#DISABLE_SYSTEM_SERVICES=""
#KEEP_SESSION_SERVICES=""
MASK_SESSION_SERVICES=""
MASK_SYSTEM_SERVICES="nginx nxserver openipmi vboxdrv vboxautostart-service vboxballoonctrl-service vboxweb-service x2goserver"
LIGHTDM_CONF="user-session=mate
greeter-hide-users=true"
DNS_SERVER="xxx.xxx.xxx.xxx xxx.xxx.xxx.xxx"
POST_INIT_LAB_CLIENT="section_lab_client"

[04:7b:cb:*]
POST_INIT_RM_BLACKLIST_AMDGPU="section_rm_blacklist_amdgpu"

[lab_client]
rm /etc/xdg/autostart/nm-applet.desktop

[rm_blacklist_amdgpu]
rm /etc/modprobe.d/blacklist-amdgpu.conf
modprobe amdgpu

![20230221_144722](https://user-images.githubusercontent.com/85257493/220445316-b1938b96-5e3e-47a1-abec-aff790e99fe1.jpg)

**Comment by alkisg** [comment]
Well, the problem is that /etc/xdg/autostart/nm-applet.desktop doesn't exist so it can't be deleted.
Change `rm /etc/xdg/autostart/nm-applet.desktop` to `rm -f /etc/xdg/autostart/nm-applet.desktop` to avoid the issue.

**Reply by mavrinac** [reply]
Hmm. I may have deleted it from the server itself to avoid the same applet appearing in X2Go or NoMachine sessions! So, that takes care of case #1. Case #2 seems to have been resolved by the fix for case #1.

Thanks for pointing out the obvious!
